cdap-security-spi/src/main/java/io/cdap/cdap/security/spi/authorization/PermissionManagerSpi.java

```diff
@@ -0,0 +1,80 @@
+ * the License.
+ */
+
+package io.cdap.cdap.security.spi.authorization;
+
+import io.cdap.cdap.api.security.AccessException;
+import io.cdap.cdap.proto.id.EntityId;
+import io.cdap.cdap.proto.security.Authorizable;
+import io.cdap.cdap.proto.security.GrantedPermission;
+import io.cdap.cdap.proto.security.Permission;
+import io.cdap.cdap.proto.security.Principal;
+
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Interface to manage {@link Permission} on entities. Depending on authorization extension can
+ * provide management access or not depending on security type. In the latter case permission must
+ * be managed using native tools for the security environment.
+ */
+public interface PermissionManagerSpi {
+
+  /**
+   * Grants a {@link Principal} authorization to perform a set of {@link Permission permissions} on
+   * {@link EntityId} represented by the {@link Authorizable} Note: this grant is used to support
+   * wildcard privilege management
+   *
+   * @param authorizable The {@link Authorizable} on which the {@link Permission} are to be
+   *     granted
+   * @param principal the {@link Principal} that performs the permissions. This could be a user,
+   *     or role
+   * @param permissions the set of {@link Permission permissions} to grant.
+   */
+  AuthorizationResult grant(Authorizable authorizable, Principal principal, Set<? extends Permission> permissions)
+      throws AccessException;
+
+  /**
+   * Revokes a {@link Principal} authorization to perform a set of {@link Permission permissions} on
+   * {@link EntityId} represented by the {@link Authorizable} Note: this revoke is used to support
+   * wildcard privilege management.
+   *
+   * @param authorizable the {@link Authorizable} whose {@link Permission permissions} are to be
+   *     revoked
+   * @param principal the {@link Principal} that performs the permissions. This could be a user,
+   *     group or role
+   * @param permissions the set of {@link Permission permissions} to revoke
+   */
+  AuthorizationResult revoke(Authorizable authorizable, Principal principal, Set<? extends Permission> permissions)
+      throws AccessException;
+
+  /**
+   * Revokes all {@link Principal}s authorization to perform any set of {@link Permission
+   * permissions} on {@link EntityId} represented by the {@link Authorizable}
+   *
+   * @param authorizable the {@link Authorizable} on which all {@link Permission permissions}
+   *     are to be revoked
+   */
+  AuthorizationResult revoke(Authorizable authorizable) throws AccessException;
+
+  /**
+   * Returns all the {@link GrantedPermission} for the specified {@link Principal}.
+   *
+   * @param principal the {@link Principal} for which to return privileges
+   * @return a {@link Set} of {@link GrantedPermission} for the specified principal
+   */
+  Map<GrantedPermission, AuthorizationResult> listGrants(Principal principal) throws AccessException;
```

Comment:
I don't think you need a map here. You are not checking caller authorization for each permission, but the whole call can be unauthorized.